### PR TITLE
Improved Output in Validation with no Entities

### PR DESCRIPTION
**Pull request by rogerwalt**
Instead of "OK - The mapping files are correct" it now says "FAIL - No
entities defined!".

Maybe someone can help me write a correct test for the
ValidateSchemaCommand, I wasn't able to provide that, mostly because
there is no test for this command yet.

### CODE REVIEW

File: lib/Doctrine/ORM/Tools/Console/Command/ValidateSchemaCommand.php [hunk 1]
```diff
@@ -76,7 +76,9 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         if ($input->getOption('skip-mapping')) {
             $output->writeln('<comment>[Mapping]  Skipped mapping check.</comment>');
-        } elseif ($errors = $validator->validateMapping()) {
+        } elseif (($errors = $validator->validateMapping()) === false) {
+            $output->writeln('<error>[Mapping]  FAIL - No entities defined!</error>');
+        } elseif (count($errors) > 0) {
```
Comment (rogerwalt) [inline lib/Doctrine/ORM/Tools/Console/Command/ValidateSchemaCommand.php]: Now by looking at this I think $error won't be defined here. I should really write a test for the `ValidateSchemaCommand`.

File: lib/Doctrine/ORM/Tools/SchemaValidator.php [hunk 2]
```diff
@@ -67,6 +67,10 @@ public function validateMapping()
         $cmf = $this->em->getMetadataFactory();
         $classes = $cmf->getAllMetadata();
 
+        if (count($classes) === 0) {
+            return false;
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Tools/SchemaValidator.php]: 1. return type is inconsistent with signature
2. an empty set of mappings is perfectly valid here

File: lib/Doctrine/ORM/Tools/Console/Command/ValidateSchemaCommand.php (hunk 1 above)
Comment (Ocramius) [inline lib/Doctrine/ORM/Tools/Console/Command/ValidateSchemaCommand.php]: I suggest counting `$em->getMetadataFactory()->getAllMetadata()` and dealing with the case where 0 mappings are found directly in the CLI command. The validator context is much more specific, and an empty set of metadata is indeed valid.

File: tests/Doctrine/Tests/ORM/Tools/SchemaValidatorTest.php [hunk 3]
```diff
@@ -24,6 +27,22 @@ public function setUp()
...
+        $config->setMetadataDriverImpl(new ZeroEntitiesMappingDriver());
+        $config->setProxyDir(__DIR__ . '/Proxies');
+        $config->setProxyNamespace('Doctrine\Tests\Proxies');
+        $em = $this->_getTestEntityManager(null, $config, null, true);
+
+        // setup new validator for $em
+        $validator = new SchemaValidator($em);
+
+        $result = $validator->validateMapping();
+        $this->assertFalse($result);
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Tools/SchemaValidatorTest.php]: Should simply be `self::assertEmpty($result)`

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Pull request closed without merging**